Source/Core/JulianDate.js

```diff
@@ -313,17 +319,21 @@ define([
                             var dayOfWeek = +tokens[3] || 0;
 
                             dashCount = date.split('-').length - 1;
```

Comment:
You have `dashCount` inside the pragmas at L279 but outside in other places.  I think it can be inside everywhere since it's only used in these `DeveloperError` `if` blocks